python/tank/context.py

```diff
@@ -668,6 +668,141 @@ def create_empty(tk):
     """
     return Context(tk)
 
+def from_entity_dictionary(tk, entity_dictionary):
+    """
+    """
+    if not isinstance(entity_dictionary, dict):
+        raise TankError("Cannot create a context from an empty or invalid entity dictionary!")
+    
+    if "type" not in entity_dictionary:
+        raise TankError("Cannot create a context without an entity type!")
+    
+    if "id" not in entity_dictionary:
+        raise TankError("Cannot create a context without an entity id!")
+    
+    # prep our context data structure
+    context = {
+        "tk": tk,
+        "project": None,
+        "entity": None,
+        "step": None,
+        "user": None,
+        "task": None,
+        "additional_entities": []
+    }
+
+    entity_type = entity_dictionary["type"]
+    entity_id = entity_dictionary["id"]
+
+    # try to determine the various entities from the entity dictionary:
+    project = entity = step = task = None
```

Comment:
While multiple assignment on a line is valid syntax, this seems ugly.

PEP8 discourages compound statements, and this actually can become a likely error if the assignment is to something mutable.

Split it into multiple lines?